numpy/core/tests/test_arrayprint.py

```diff
@@ -182,6 +182,16 @@ def test_structure_format(self):
                 (1., 2.1234567890123456789, 3.), dtype=('f8,f8,f8'))
         assert_equal(np.array2string(array_scalar), "( 1.,  2.12345679,  3.)")
 
+    def test_unstructured_void_repr(self):
+        a = np.array([27, 91, 50, 75,  7, 65, 10,  8,
+                      27, 91, 51, 49,109, 82,101,100], dtype='u1').view('V8')
+        assert_equal(repr(a[0]), r"void(b'\x1B\x5B\x32\x4B\x07\x41\x0A\x08')")
+        assert_equal(str(a[0]), r"b'\x1B\x5B\x32\x4B\x07\x41\x0A\x08'")
+        assert_equal(repr(a),
+            r"array([b'\x1B\x5B\x32\x4B\x07\x41\x0A\x08'," "\n"
+            r"       b'\x1B\x5B\x33\x31\x6D\x52\x65\x64']," "\n"
+            r"      dtype='|V8')")
+
```

Comment:
Can we add an `eval(repr(x), vars(np)) == x` test, like we have elsewhere, for both scalars and arrays? That's one of the features of this patch too